lib/assert.js

```diff
@@ -608,14 +613,169 @@ assert.notStrictEqual = function notStrictEqual(actual, expected, message) {
+    return false;
+  }
+
+  // Check for Map object equality
+  if (actual instanceof Map && expected instanceof Map) {
+    if (actual.size !== expected.size) return false;
+    for (const { 0: key, 1: val } of actual) {
+      if (!expected.has(key)) return false;
+      if (!compareBranch(val, expected.get(key), loose, comparedObjects))
+        return false;
+    }
+    return true;
+  }
+
+  // Check for Set object equality
+  if (actual instanceof Set && expected instanceof Set) {
+    if (actual.size !== expected.size) return false;
+    for (const item of actual) {
+      if (!expected.has(item)) return false;
+    }
+    return true;
+  }
+
+  // Use Reflect.ownKeys() instead of Object.keys() to include symbol properties
+  const keysActual = ReflectOwnKeys(actual);
+  const keysExpected = ReflectOwnKeys(expected);
+
+  // Check if number of keys are equal
+  if (keysActual.length !== keysExpected.length) {
+    return false;
+  }
+
+  // Handle circular references
+  if (comparedObjects.has(actual)) {
+    return true;
+  }
+  comparedObjects.add(actual);
+
+  // Check if all keys and values recursively match
+  for (const key of keysActual) {
+    if (
+      !keysExpected.includes(key) ||
+      !compareBranch(actual[key], expected[key], loose, comparedObjects)
+    ) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+/**
+ * The strict equivalence assertion test between two objects
+ * @param {any} actual
+ * @param {any} expected
+ * @param {string | Error} [message]
+ * @returns {void}
+ */
+assert.matchObjectStrict = function matchObjectStrict(
+  actual,
+  expected,
+  message,
+) {
+  if (arguments.length < 2) {
+    throw new ERR_MISSING_ARGS('actual', 'expected');
+  }
+
+  if (!compareBranch(actual, expected)) {
+    innerFail({
+      actual,
+      expected,
+      message,
+      operator: 'matchObjectStrict',
+      stackStartFn: matchObjectStrict,
+    });
+  }
+};
+
+/**
+ * The equivalence assertion test between two objects
+ * @param {any} actual
+ * @param {any} expected
+ * @param {string | Error} [message]
+ * @returns {void}
+ */
+assert.matchObject = function matchObject(actual, expected, message) {
+  if (arguments.length < 2) {
+    throw new ERR_MISSING_ARGS('actual', 'expected');
+  }
+
+  if (!compareBranch(actual, expected, false)) {
+    innerFail({
+      actual,
+      expected,
+      message,
+      operator: 'matchObject',
+      stackStartFn: matchObject,
+    });
+  }
+};
+
 class Comparison {
   constructor(obj, keys, actual) {
     for (const key of keys) {
       if (key in obj) {
-        if (actual !== undefined &&
-            typeof actual[key] === 'string' &&
-            isRegExp(obj[key]) &&
-            RegExpPrototypeExec(obj[key], actual[key]) !== null) {
+        if (
+          actual !== undefined &&
+          typeof actual[key] === 'string' &&
+          isRegExp(obj[key]) &&
+          RegExpPrototypeExec(obj[key], actual[key]) !== null
+        ) {
```

Comment:
Still some over there
```suggestion
        if (actual !== undefined &&
            typeof actual[key] === 'string' &&
            isRegExp(obj[key]) &&
            RegExpPrototypeExec(obj[key], actual[key]) !== null) {
```